File: c++/007-ReverseInteger.cpp
#include<cmath>
#include<iostream>
using namespace std;

class Solution {
public:
    int reverse(int x) {
        long long origin = abs(x);
        long long ans = 0;
        bool negative = (x < 0);
        while (origin > 0) {
          ans = ans * (long long int)10 + origin % 10;
          origin /= 10;
        }
        
        // According to the test cases, if the result is exceeding the limit of int-32, should return 0.
        if (negative) {
            ans = -ans;
			if (-((long long)1 << 31) > ans) ans = 0;
        } else {
			if (((long long)1 << 31) - 1 < ans) ans = 0;
        }
        return ans;
    }
};

int main() {
	return 0;
}

File: python/007-ReverseInteger.py
class Solution:
    # @param {integer} x
    # @return {integer}
    def reverse(self, x):
        res, tmp_x = 0, abs(x)
        while tmp_x > 0 :            
            res = res * 10 + tmp_x % 10
            tmp_x = int(tmp_x / 10)
        if x < 0 :
            return -res
        else : 
            return res
            
solution = Solution()
x = 123
print(solution.reverse(x))
x = -123
print(solution.reverse(x))
